lab1/main_test.js

```diff
@@ -3,21 +3,36 @@ const assert = require('assert');
 const { MyClass, Student } = require('./main');
 
 test("Test MyClass's addStudent", () => {
-    // TODO
-    throw new Error("Test not implemented");
+    const student = new Student();
+    const myClass = new MyClass();
+
+    assert.strictEqual(myClass.addStudent(student), 0);         // first student
+    assert.strictEqual(myClass.addStudent(myClass), -1);        // return  -1
 });
 
 test("Test MyClass's getStudentById", () => {
-    // TODO
-    throw new Error("Test not implemented");
+    const student = new Student();
+    const myClass = new MyClass();
+
+    const newStudentId = myClass.addStudent(student);
+    assert.strictEqual(myClass.getStudentById(newStudentId), student);
+    assert.strictEqual(myClass.getStudentById(100), null);                     //  over length 
 });
 
 test("Test Student's setName", () => {
-    // TODO
-    throw new Error("Test not implemented");
+    const student = new Student();
+
+    student.setName("John");   // 使用assert.strictEqual来断言student的name属性是否被正确设置
+    assert.strictEqual(student.name, "John");
+
+    student.setName(123);      // 尝试使用非字符串值设置
```

Comment:
Are those comments and code written on your own? If it's copied or generated from somewhere else, it's okay. But I hope you can understand and encourage you to write them by yourself.